unison-cli/src/Unison/CommandLine/Globbing.hs

```diff
@@ -107,11 +105,10 @@ expandGlobs ::
   -- | Nothing if arg was not a glob.
   -- otherwise, fully expanded, absolute paths. E.g. [".base.List.map"]
   Maybe [String]
-expandGlobs targets rootBranch currentPath s = either recover Just $ do
+expandGlobs targets rootBranch currentPath s = do
+  guard (not . null $ targets)
```

Comment:
Moved this up to the top, no sense in doing any work if we don't have anything to glob.

I also switched this block to `Maybe` instead of `Either` since we only have one failure mode now.